### PR TITLE
 add rest api to get tenants by state

### PR DESCRIPTION
**Pull request by dushansilva**
This PR will add a REST API to the store to retrieve tenants by its state.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java [hunk 1]
```diff
@@ -4586,6 +4587,20 @@ public static Cache getAPIContextCache() {
         return tenantDomains;
     }
 
+    public static Set<String> getTenantDomainsByState(String state) throws UserStoreException {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: Shall we add method comments

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/TenantsApiServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import javax.ws.rs.core.Response;
+
+import static org.wso2.carbon.apimgt.impl.APIConstants.TENANT_STATE_ACTIVE;
+import static org.wso2.carbon.apimgt.impl.APIConstants.TENANT_STATE_INACTIVE;
+
+public class TenantsApiServiceImpl implements TenantsApiService {
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/TenantsApiServiceImpl.java]: Let's add the class comments

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/TenantsApiServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        try {
+            List<TenantDTO> tenantDTOList = new ArrayList<>();
+            Integer paginationLimit = limit != null ? limit : RestApiConstants.PAGINATION_LIMIT_DEFAULT;
+            Integer paginationOffset = offset != null ? offset : RestApiConstants.PAGINATION_OFFSET_DEFAULT;
+            PaginationDTO paginationDTO = new PaginationDTO();
+            paginationDTO.setOffset(paginationOffset);
+            paginationDTO.setLimit(paginationLimit);
+
+            if (!state.equalsIgnoreCase(TENANT_STATE_ACTIVE) && !state.equalsIgnoreCase(TENANT_STATE_INACTIVE)) {
+                TenantListDTO tenantList = new TenantListDTO();
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/TenantsApiServiceImpl.java]: Is this handling an invalid state? Shouldn't we just return a 400 saying invalid state and only these states are allowed?

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml [hunk 4]
```diff
@@ -3172,6 +3172,68 @@ paths:
...
+  # The tenant resource APIs
+  ######################################################
+  '/tenants':
+    #--------------------------------------------------------
+    # Retrieve tenants by state
+    #--------------------------------------------------------
+    get:
+      security:
+        - OAuth2Security: []
+      x-wso2-curl: "curl -X GET \"https://localhost:9443/api/am/store/v1.0/tenants?state=active\" -H \"Authorization: Basic 3a7ecd76-4856-33ce-a703-e23e6cd24944\""
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml]: Let's add examples as mentioned in https://github.com/wso2/carbon-apimgt/wiki/Adding-Examples-for-REST-APIs

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml [hunk 5]
```diff
@@ -3172,6 +3172,68 @@ paths:
...
+          type: string
+          required: false
+          default: 'active'
+          enum:
+            - active
+            - inactive
+        - $ref: '#/parameters/limit'
+        - $ref: '#/parameters/offset'
+      tags:
+        - active tenants
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml]: We can simplify the tag as "Tenants"

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml [hunk 6]
```diff
@@ -3332,7 +3394,7 @@ parameters:
     name: labelName
     in: query
     description: |
-      Name of the API microgateway label.
+      Name of the API microgateway label.s
```
Comment (malinthaprasan) [inline components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml]: typo

File: components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java (hunk 1 above)
Comment (dushansilva) [reply components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/APIUtil.java]: fixed https://github.com/wso2/carbon-apimgt/pull/6811/commits/a840167e827f482f100d25dadea5b4a46dc53be1

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/TenantsApiServiceImpl.java (hunk 2 above)
Comment (dushansilva) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/TenantsApiServiceImpl.java]: fixed https://github.com/wso2/carbon-apimgt/pull/6811/commits/a840167e827f482f100d25dadea5b4a46dc53be1

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/TenantsApiServiceImpl.java (hunk 3 above)
Comment (dushansilva) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/store/v1/impl/TenantsApiServiceImpl.java]: fixed https://github.com/wso2/carbon-apimgt/pull/6811/commits/a840167e827f482f100d25dadea5b4a46dc53be1

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml (hunk 4 above)
Comment (dushansilva) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml]: fixed https://github.com/wso2/carbon-apimgt/pull/6811/commits/a840167e827f482f100d25dadea5b4a46dc53be1

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml (hunk 5 above)
Comment (dushansilva) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml]: fixed https://github.com/wso2/carbon-apimgt/pull/6811/commits/a840167e827f482f100d25dadea5b4a46dc53be1

File: components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml (hunk 6 above)
Comment (dushansilva) [reply components/apimgt/org.wso2.carbon.apimgt.rest.api.store.v1/src/main/resources/store-api.yaml]: fixed https://github.com/wso2/carbon-apimgt/pull/6811/commits/a840167e827f482f100d25dadea5b4a46dc53be1

**Pull request merged**